github-actions/feature-request/src/octokit.ts

```diff
@@ -0,0 +1,260 @@
+    issue: Octokit.IssuesGetResponse | Octokit.SearchIssuesAndPullRequestsResponseItemsItem,
+  ): Issue {
+    return {
+      author: { name: issue.user.login, isGitHubApp: issue.user.type === 'Bot' },
+      body: issue.body,
+      number: issue.number,
+      title: issue.title,
+      labels: (issue.labels as Octokit.IssuesGetLabelResponse[]).map((label) => label.name),
+      open: issue.state === 'open',
+      locked: (issue as any).locked,
+      numComments: issue.comments,
+      reactions: (issue as any).reactions,
+      assignee: issue.assignee?.login ?? (issue as any).assignees?.[0]?.login,
+      createdAt: +new Date(issue.created_at),
+      updatedAt: +new Date(issue.updated_at),
+      closedAt: issue.closed_at ? +new Date((issue.closed_at as unknown) as string) : undefined,
+    }
+  }
+
+  async repoHasLabel(name: string): Promise<boolean> {
+    try {
+      await this.octokit.issues.getLabel({ ...this.params, name })
+      return true
+    } catch (err) {
+      if (err.status === 404) {
+        return this.options.readonly && this.mockLabels.has(name)
+      }
+      throw err
+    }
+  }
+}
+
+export class OctoKitIssue extends OctoKit implements GitHubIssueAPI {
+  constructor(
+    token: string,
+    protected params: { repo: string; owner: string },
+    private issueData: { number: number } | Issue,
+    options: { readonly: boolean } = { readonly: false },
+  ) {
+    super(token, params, options)
+    log('running bot on issue', issueData.number)
+  }
+
+  async close(): Promise<void> {
+    log('Closing issue ' + this.issueData.number)
+    if (!this.options.readonly)
+      await this.octokit.issues.update({
+        ...this.params,
+        issue_number: this.issueData.number,
+        state: 'closed',
+      })
+  }
+
+  async get(): Promise<Issue> {
+    if (isIssue(this.issueData)) {
+      log('Got issue data from query result ' + this.issueData.number)
+      return this.issueData
+    }
+
+    log('Fetching issue ' + this.issueData.number)
+    const issue = (
+      await this.octokit.issues.get({
+        ...this.params,
+        issue_number: this.issueData.number,
+        mediaType: { previews: ['squirrel-girl'] },
+      })
+    ).data
+    return (this.issueData = this.octokitIssueToIssue(issue))
+  }
+
+  async postComment(body: string): Promise<void> {
+    log(`Posting comment on ${this.issueData.number}`)
+    if (!this.options.readonly)
+      await this.octokit.issues.createComment({
+        ...this.params,
+        issue_number: this.issueData.number,
+        body,
+      })
+  }
+
+  async deleteComment(id: number): Promise<void> {
+    log(`Deleting comment ${id} on ${this.issueData.number}`)
+    if (!this.options.readonly)
+      await this.octokit.issues.deleteComment({
+        owner: this.params.owner,
+        repo: this.params.repo,
+        comment_id: id,
+      })
+  }
+
+  async *getComments(last?: boolean): AsyncIterableIterator<Comment[]> {
```

Comment:
Can we just yield the individual comments?  I suppose this is a comment on the general pattern as well now that I am looking in more detail.

It seems to me that since often when we request things, because have to iterate through them to make the localized version (i.e. `Octokit.IssuesListCommentsResponseItem` is converted into `Comment`), that we should just yield out the Comment directly. I feel like this is much easier to reason about in its usages because then. `getComents()` moves from returning what amounts to `Comment[][]` to `Comment[]`.

I am not sure if I am missing something that makes the current method more valuable.